rend3/src/managers/mesh.rs

```diff
@@ -229,6 +229,95 @@ impl MeshManager {
         self.registry.insert(handle, mesh);
     }
 
+    /// Duplicates a mesh's vertex data so that it can be skinned on the GPU.
+    pub fn allocate_skeleton_mesh(
+        &mut self,
+        device: &Device,
+        _queue: &Queue,
+        encoder: &mut CommandEncoder,
+        mesh_handle: &MeshHandle,
+    ) -> Range<usize> {
+        // Need to fetch internal data twice, because the returned mesh borrows &self
+        let needed_verts = self.internal_data(mesh_handle.get_raw()).vertex_range.len();
+        let vertex_range = match self.vertex_alloc.allocate_range(needed_verts) {
+            Ok(range) => range,
+            Err(_) => {
+                self.reallocate_buffers(device, encoder, needed_verts as u32, 0);
+                self.vertex_alloc
+                    .allocate_range(needed_verts)
+                    .expect("We just reallocated")
+            }
+        };
+
+        let original = self.internal_data(mesh_handle.get_raw());
+
+        encoder.copy_buffer_to_buffer(
+            &self.buffers.vertex_position,
+            (original.vertex_range.start * VERTEX_POSITION_SIZE) as u64,
+            &self.buffers.vertex_position,
+            vertex_range.start as u64,
```

Comment:
Have the same problem here :)

Reply:
We're lucky that at least someone is paying attention here... :sweat_smile: Thanks again